File: Backend/src/main/java/com/AlaCartApp/models/entity/Product.java
package com.AlaCartApp.models.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Entity
@Data
@Table(name = "products")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;
    @Column(name = "nombre", nullable = false)
    private String name;
    @Column(name = "precio", nullable = false)
    private Double price;
    @ManyToOne()
    @JoinColumn(name = "category_id", referencedColumnName = "id")
    private Category category;
    @ManyToOne()
    @JoinColumn(name = "sub_category_id", referencedColumnName = "id")
    private SubCategory subCategory;
    @Column(name = "description", nullable = false)
    private String description;
    @Column(name = "estado" , nullable = false)
    private Boolean state;
    //@OneToMany(mappedBy = "product")
    //private List<Topics> topics;
    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
    @JoinColumn(name="product_id")
    private List<Image> images;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    List<Qualification> qualifications;
}

File: Frontend/src/types/Product.ts
// "Platos" | "Postres" | "Bebidas";

type Category = {
  id: string
  name: string
  available: boolean
}

type Image = {
  id: string
  imageUrl: string
}

type SubCategory = {
  id: string
  name: string
  available: boolean
}

export interface Product {
  id: string
  name: string
  price: number
  category: Category
  description: string
  images: Image[]
  state: boolean
  subCategory: SubCategory
  rating?: number
  optionals?: string[]
}

export type NonEssentialsProduct = Omit<
  Product,
  'id' | 'description' | 'subCategory' | 'optionals' | 'state'
>
